JSON/JSON.sublime-syntax

```diff
@@ -110,28 +121,31 @@ contexts:
                   set:
                     - match: ','
                       scope: punctuation.separator.mapping.pair.json
-                      pop: true
+                      pop: 1
                     - match: \s*(?=\})
-                      pop: true
+                      pop: 1
                     - match: \s(?!/[/*])(?=[^\s,])|[^\s,]
                       scope: invalid.illegal.expected-mapping-separator.json
-                      pop: true
+                      pop: 1
         - match: '[^\s\}]'
           scope: invalid.illegal.expected-mapping-key.json
+
   string:
-    - match: '"'
+    - match: \"
       scope: punctuation.definition.string.begin.json
       push: inside-string
+
   inside-string:
     - meta_scope: string.quoted.double.json
     - meta_include_prototype: false
-    - match: '"'
+    - match: \"
       scope: punctuation.definition.string.end.json
-      pop: true
+      pop: 1
     - include: string-escape
-    - match: $\n?
+    - match: \n
```

Comment:
I guess these two behave different at EOF but probably doesn't matter?

Reply:
1. We don't need to pop a string context at eof, because there is nothing which would follow.
2. Strings are leaf-contexts. They don't include itself, such as groups can. So consuming the `\n` is safe (not causing any recursive context not to pop. _All other contexts which include strings would need to make sure to match `$`._

These are the two major points I use to choose whether to `$\n?` or `\n`. As both conditions are met the latter one is safe to be used.